inc/Modules/Plugin/Cache.php

```diff
@@ -152,24 +155,26 @@ public static function build_plugins_transient(): array|\WP_Error {
 		$updates = get_plugin_updates();
 		if ( ! empty( $updates ) ) {
 			foreach ( $updates as $slug => $update ) {
-				if ( isset( $plugins[ $slug ] ) ) {
-					$plugins[ $slug ]['update'] = $update->update;
+				if ( ! isset( $plugins[ $slug ] ) ) {
+					continue;
 				}
+
+				$plugins[ $slug ]['update'] = $update->update;
```

Comment:
Check property_exists() first, lots of plugins like to mess with their own/other plugin payload props